CougSat1-Ground/src/space/cougs/ground/gui/CougSat1GUI.java

```diff
@@ -0,0 +1,232 @@
+
+import java.awt.BorderLayout;
+import java.awt.CardLayout;
+import java.awt.Component;
+import java.awt.Container;
+import java.awt.Font;
+import java.awt.GridLayout;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+
+import javax.swing.AbstractButton;
+import javax.swing.BorderFactory;
+import javax.swing.ImageIcon;
+import javax.swing.JButton;
+import javax.swing.JPanel;
+import javax.swing.SwingConstants;
+
+import space.cougs.ground.gui.subsystems.ADCS;
+import space.cougs.ground.gui.subsystems.Camera;
+import space.cougs.ground.gui.subsystems.Comms;
+import space.cougs.ground.gui.subsystems.EPS;
+import space.cougs.ground.gui.subsystems.Health;
+import space.cougs.ground.gui.subsystems.IFJR;
+import space.cougs.ground.gui.subsystems.IHU;
+import space.cougs.ground.gui.subsystems.Plant;
+import space.cougs.ground.gui.subsystems.SatelliteInfo;
+import space.cougs.ground.gui.utils.CustomColors;
+import space.cougs.ground.gui.utils.Fonts;
+import space.cougs.ground.satellites.CougSat;
+
+class CougSat1GUI extends JPanel implements UIScaling, SatelliteInfo {
+
+	private static final long serialVersionUID = 1L;
+
+	private final Health health; // all telemetry data
+	private final ADCS adcs;
+	private final IFJR ifjr;
+	private final IHU ihu; // Internal houskeeping unit shhhh
+	private final Comms comms;
+	private final Camera camera; // camera data
+	private final Plant plant; // images of plant
+	private final EPS eps; // power?
+
+	JPanel subSystemWrapper = new JPanel(new CardLayout());
+
+	CardSwitcher cardSwitcher;
+
+	public CougSat1GUI() {
+		super();
+
+		this.setLayout(new BorderLayout());
+
+		health = new Health();
+		adcs = new ADCS();
+		ifjr = new IFJR();
+		ihu = new IHU();
+		comms = new Comms();
+		camera = new Camera();
+		plant = new Plant();
+		eps = new EPS();
+
+		subSystemWrapper.add(health, "Health");
+		subSystemWrapper.add(adcs, "Attitude");
+		subSystemWrapper.add(ifjr, "IFJR");
+		subSystemWrapper.add(ihu, "Computer");
+		subSystemWrapper.add(comms, "Radio");
+		subSystemWrapper.add(camera, "Camera");
+		subSystemWrapper.add(plant, "Plant");
+		subSystemWrapper.add(eps, "Power");
+		subSystemWrapper.setOpaque(false);
+		this.add(subSystemWrapper, BorderLayout.CENTER);
+
+		cardSwitcher = new CardSwitcher();
+		this.add(cardSwitcher, BorderLayout.LINE_START);
+
+		this.setBackground(CustomColors.BACKGROUND1);
+		// this.setBorder(BorderFactory.createEmptyBorder(10, 10, 10, 10));
+
+	}
+
+	private class CardSwitcher extends JPanel implements ActionListener, UIScaling {
+
+		private static final long serialVersionUID = 1L;
+
+		private final JButton healthButton;
+		private final JButton adcsButton;
+		private final JButton ifjrButton;
+		private final JButton ihuButton;
+		private final JButton commsButton;
+		private final JButton cameraButton;
+		private final JButton plantButton;
+		private final JButton epsButton;
+
+		CardSwitcher() {
+
+			healthButton = new JButton("Health");
```

Comment:
Can CardSwitcher private class use a private constructor
Move constructor to instantiation